### PR TITLE
Use full path in #includes

### LINKED ISSUE #283: use full path for header files
This PR is about removing the lua prefix from file names. You just happen to spot another problem that needs to be fixed...

_Originally posted by @andrsd in https://github.com/Open-Sn/opensn/pull/281#discussion_r1623398987_

### PR DESCRIPTION
**Pull request by andrsd**
- Use full relative path from the repo root in #includes
- Don't include `lua` dir into include dir search

Closes #283

### CODE REVIEW

File: lua/CMakeLists.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ target_include_directories(libopensnlua
     $<INSTALL_INTERFACE:include/opensn>
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
     $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
-    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${PROJECT_SOURCE_DIR}
```
Comment (andrsd) [inline lua/CMakeLists.txt:10]: The reason for this change is that this added `<repo_root>/lua` into include path, so people could omit the `lua` prefix in the `#include`s. After this change they have to use the path from the repo_root, i.e. include the `lua` subdirectory.

**Review by zhardy-lanl — APPROVED** [review]
The only comment I have here is that in general I would be okay with using relative paths for *only* the associated .h file.

**Review by wdhawkins — APPROVED** [review]

**Comment by andrsd** [comment]
> The only comment I have here is that in general I would be okay with using relative paths for _only_ the associated .h file.

This was exactly the case that put all this into motion :-)

image

And the [discussion](https://github.com/Open-Sn/opensn/pull/281#pullrequestreview-2089526484) that triggered this.

**Review by ragusa — APPROVED** [review]
I know it makes it feel pedantic but I think it avoids any confusion.

**Pull request merged**